### PR TITLE
added nextjs 13 minimal starter

### PR DESCRIPTION
**Pull request by gustavocodess**
## Description

Added minimal starter for next js version 13 integrated with Builder.io space.

### CODE REVIEW

File: examples/simple-next-js-13/src/app/[[...slug]]/ClientComponent.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import builderConfig from '../../../builderConfig.json';
+
+builder.init(builderConfig.apiKey);
+
+export default function ClientComponent({ children, builderContent }: any) {
+  return (
+    <>
+      {children}
+      <br />
+      <p>This section is rendered client side</p>
```
Comment (teleaziz) [inline examples/simple-next-js-13/src/app/[[...slug]]/ClientComponent.tsx:12]: That's not accurate btw

File: examples/simple-next-js-13/src/app/[[...slug]]/ClientComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+      <BuilderComponent model="page" content={builderContent} />
```
Comment (teleaziz) [inline examples/simple-next-js-13/src/app/[[...slug]]/ClientComponent.tsx:13]: maybe revise the name/content of this file:

```
export default function BuilderPage({ builderContent }:  BuilderPageProps  => <BuilderComponent model="page" content={builderContent} />


```

File: examples/simple-next-js-13/src/app/api/hello/route.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export async function GET(request: Request) {
+  return new Response('Hello, Next.js!');
```
Comment (teleaziz) [inline examples/simple-next-js-13/src/app/api/hello/route.ts:2]: not needed

File: examples/simple-next-js-13/src/app/not-builder/page.tsx [hunk 4]
```diff
@@ -0,0 +1,7 @@
+export default async function Home() {
+  return (
+    <main>
+      <h3>Home page sample of page not published on builder</h3>
```
Comment (teleaziz) [inline examples/simple-next-js-13/src/app/not-builder/page.tsx:4]: Please make it a catch all route, like the other starters

File: examples/simple-next-js-13/src/app/page.module.css [hunk 5]
```diff
@@ -0,0 +1,271 @@
+.main {
+  display: flex;
```
Comment (teleaziz) [inline examples/simple-next-js-13/src/app/page.module.css:2]: probably not needed once moved to catchall route

File: examples/simple-next-js-13/src/app/[[...slug]]/ClientComponent.tsx (hunk 1 above)
Comment (teleaziz) [reply examples/simple-next-js-13/src/app/[[...slug]]/ClientComponent.tsx:12]: to elaborate, client components can be rendered on the server but will be rehydrated on the client while server components only run on the server

File: examples/simple-next-js-13/public/next.svg [hunk 6]
```diff
@@ -0,0 +1 @@
...55v12.6H80.5V0h68.7zm34.3 0h-17.8l63.8 79.4h17.9l-32-39.7 32-39.6h-17.9l-23 28.6-23-28.6zm18.3 56.7-9-11-27.1 33.7h17.8l18.3-22.7z"/><path fill="#000" d="M81 79.3 17 0H0v79.3h13.6V17l50.2 62.3H81Zm252.6-.4c-1 0-1.8-.4-2.5-1s-1.1-1.6-1.1-2.6.3-1.8 1-2.5 1.6-1 2.6-1 1.8.3 2.5 1a3.4 3.4 0 0 1 .6 4.3 3.7 3.7 0 0 1-3 1.8zm23.2-33.5h6v23.3c0 2.1-.4 4-1.3 5.5a9.1 9.1 0 0 1-3.8 3.5c-1.6.8-3.5 1.3-5.7 1.3-2 0-3.7-.4-5.3-1s-2.8-1.8-3.7-3.2c-.9-1.3-1.4-3-1.4-5h6c.1.8.3 1.6.7 2.2s1 1.2 1.6 1.5c.7.4 1.5.5 2.4.5 1 0 1.8-.2 2.4-.6a4 4 0 0 0 1.6-1.8c.3-.8.5-1.8.5-3V45.5zm30.9 9.1a4.4 4.4 0 0 0-2-3.3 7.5 7.5 0 0 0-4.3-1.1c-1.3 0-2.4.2-3.3.5-.9.4-1.6 1-2 1.6a3.5 3.5 0 0 0-.3 4c.3.5.7.9 1.3 1.2l1.8 1 2 .5 3.2.8c1.3.3 2.5.7 3.7 1.2a13 13 0 0 1 3.2 1.8 8.1 8.1 0 0 1 3 6.5c0 2-.5 3.7-1.5 5.1a10 10 0 0 1-4.4 3.5c-1.8.8-4.1 1.2-6.8 1.2-2.6 0-4.9-.4-6.8-1.2-2-.8-3.4-2-4.5-3.5a10 10 0 0 1-1.7-5.6h6a5 5 0 0 0 3.5 4.6c1 .4 2.2.6 3.4.6 1.3 0 2.5-.2 3.5-.6 1-.4 1.8-1 2.4-1.7a4 4 0 0 0 .8-2.4c0-.9-.2-1.6-.7-2.2a11 11 0 0 0-2.1-1.4l-3.2-1-3.8-1c-2.8-.7-5-1.7-6.6-3.2a7.2 7.2 0 0 1-2.4-5.7 8 8 0 0 1 1.7-5 10 10 0 0 1 4.3-3.5c2-.8 4-1.2 6.4-1.2 2.3 0 4.4.4 6.2 1.2 1.8.8 3.2 2 4.3 3.4 1 1.4 1.5 3 1.5 5h-5.8z"/></svg>
```
Comment (teleaziz) [inline examples/simple-next-js-13/public/next.svg:1]: are those assets used anywhere?

File: examples/next-app-directory/package.json [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+  "private": true,
+  "scripts": {
+    "dev": "next dev",
+    "build": "next build",
+    "start": "next start",
+    "lint": "next lint"
+  },
+  "dependencies": {
+    "@builder.io/react": "^2.0.15",
+    "@builder.io/sdk": "^1.1.33",
```
Comment (teleaziz) [inline examples/next-app-directory/package.json:13]: `@builder.io/sdk ` don't need to be a dependency here, it's already included in `@builder.io/react`, it caused an issue for me today testing the remix starter and ended up removing it to get the latest updates

Comment (teleaziz) [reply examples/next-app-directory/package.json:13]: https://github.com/BuilderIO/builder/commit/0997d07ad1cfe366b5ff3442149efde697ddb43a#diff-485037290c6f6b4181e1aae71bd2d5da9c25254675a04753764e0c092cac12f1R13

Comment (gustavocodess) [reply examples/next-app-directory/package.json:13]: I am using to get the content types on the ClientComponet file:

https://github.com/BuilderIO/builder/pull/1890/files#diff-374bce45b54565c4c1dc88f611c01ca689e1772670ebed8b024e73a3fd0a3e4d

**Review by teleaziz — APPROVED** [review]

**Pull request merged**